client/vault/smoke_test.py

```
@@ -0,0 +1,1078 @@
+async def test_view_methods(config: SmokeTestConfig):
+    """Test all view methods return valid data against a live vault.
+
+    This is a read-only test that doesn't require funded accounts or signing.
+    It uses a dummy credential since VaultClient requires one, but only makes
+    view calls.
+    """
+    # Create a client with a dummy credential (we only make view calls)
+    # Note: We need a credential to create VaultClient, but view calls don't use it
+    dummy_cred = KeyCredential(
+        account_id=AccountId("dummy.testnet"),
+        secret_key="ed25519:3D4YudUQRE39Lc4JHghuB5WM8kbgDDa34mnrEP5DdTApVH81af3e7MvFrog1CMNn67PCi6eC8x9TgDxCV9ySeGir",
```

Ruff (0.15.1)

[error] Possible hardcoded password assigned to argument: "secret_key"

(S106)